src/components/views/auth/signUp/SignUp.js

```diff
@@ -18,6 +20,16 @@ const SignUp = (props) => {
     password: '',
     confirmPassword: '',
   });
+  useEffect(() => {
+    const { location, socialSignOn, history } = props;
+    const tokenString = location && location.search;
+    if (tokenString) {
+      const token = getToken(tokenString);
+      socialSignOn(token);
+      history.push('/');
```

Comment:
What happens if `socialSignOn(token)` fails, will the user still be redirected to `/`?